experiments/spotify/3_devlop_own_source/README.md

```diff
@@ -0,0 +1,80 @@
+# Developing a pipeline with a custom (own) source
+This is example of a pipeline that includes custom source. The pipeline will be developed interatively by letting the user to modify schemas and run it again to apply changes.
+At the end we'll consume a few custom sources to show how pipeline with many sources could work.
+
+## Pipeline init
+
+Please read READMEs for `no_code_consume_source` and `consume_source` on options to initialize pipeline. In this case we do
+```
+dlt init custom-source spotify_pipeline
+```
+
+This will create `spotify_pipeline.py` configured for working with schemas iteratively. I also propose a `recreate_tables` config param that makes sure that tables are re-created with each run. (it may be also part of `load` method).
+
+## Custom source code
+
+This is exactly what we had in spotify demo.
+1. First we generate the data
+2. Then we pass the data to `extract` method. The extract method takes additional parameters that let you define the schema for the passed data.
+
+Examples:
+
+- table name. if not provided: `table` will be used
+- write_disposition, merge_keys, columns=[...], parent, - you can define a complete table schema here
+- schema - string or object: you can pass a ready schema or a schema name in whcih tables will be created
+
+## Working iteratively on schema
+
+1. you start with empty `schemas` folder
+2. you provide sensible and minimal setting for your tables into `extract` method
+3. you run your script
+4. in `schemas` folder you see inferred schemas: from your hints in `extract` and from the data itself
+5. modify the schemas and run your script again
+6. remove the `dry_run` option to actually load data
```

Comment:
nothing else comes to my mind right now

Reply:
> point 5 is where stuff will break because it's a schema and IDK what my options are.
> 
> So we need to design a good way to recover - either we auto create a backup we can restore, or offer to overwrite the broken schema. If most of the iteration happens in the extract method, it's easier to test and run iteratively IMO.

see comment - I ask exactly for this behaviour in the last comment, see our dev channel